baserun/instrumentation/instrumented_wrapper.py

```diff
@@ -0,0 +1,231 @@
+        if not parent_span.is_recording() and not Baserun.current_test_suite:
+            run = Baserun.get_or_create_current_run(
+                name="untraced",
+                trace_type=Run.RunType.RUN_TYPE_PRODUCTION,
+            )
+            parent_span = tracer.start_span(
+                f"baserun.parent.untraced",
+                kind=SpanKind.CLIENT,
+                attributes={SpanAttributes.BASERUN_RUN: Baserun.serialize_run(run)},
+            )
+
+        span = tracer.start_span(
+            **setup_span(span_name=span_name, parent_span=parent_span)
+        )
+
+        auto_end_span = True
+        try:
+            # Activate the span in the current context, but don't end it automatically
+            with trace.use_span(span, end_on_exit=False):
+                # Capture request attributes
+                set_request_attributes(
+                    instrumentor=instrumentor, span=span, kwargs=kwargs
+                )
+
+                # Actually call the wrapped method
+                try:
+                    response = wrapped_fn(*args, **kwargs)
+
+                    span.set_status(Status(StatusCode.OK))
+                except Exception as e:
+                    span.set_status(
+                        Status(
+                            description=str(e),
+                            status_code=StatusCode.ERROR,
+                        )
+                    )
+                    raise e
+
+                # If this is a streaming response, wrap it, so we can capture each chunk
+                if isinstance(response, Iterator):
+                    wrapped_response = instrumentor.generator_wrapper(response, span)
+                    # The span will be ended inside the generator once it's finished
+                    auto_end_span = False
+                    return wrapped_response
+
+                handle_response(instrumentor, span, response)
+        finally:
+            if auto_end_span:
+                if span.is_recording():
+                    span.end()
+                if parent_span.is_recording() and not Baserun.current_test_suite:
+                    parent_span.end()
+
+        return response
+
+    return instrumented_function
+
+
+def setup_span(span_name: str, parent_span: _Span) -> dict:
+    request_type = span_name.split(".")[-1]
+    parent_span.set_attribute(SpanAttributes.LLM_REQUEST_TYPE, request_type)
+
+    return {
+        "name": f"baserun.{span_name}",
+        "kind": SpanKind.CLIENT,
+        "attributes": parent_span.attributes,
+    }
+
+
+def set_request_attributes(
+    instrumentor: "BaseInstrumentor", span: _Span, kwargs: dict
+) -> _Span:
+    try:
+        instrumentor.set_request_attributes(span, kwargs)
+
+        span.set_attribute(SpanAttributes.LLM_REQUEST_MODEL, kwargs.get("model"))
+
+        max_tokens = kwargs.get("max_tokens", kwargs.get("max_tokens_to_sample"))
```

Comment:
This feels a bit inconsistent. We're also setting this in anthropic.py but not in openai.py. Can follow-up but would be nice to be a bit more deliberate

Reply:
Same with LLM_REQUEST_MODEL

Reply:
This was because i was treating openai as the "default" and making anthropic override the default where the kwarg names are different. But yeah it makes sense to have them be symmetrical.